docs/bugs.json

```diff
@@ -1,4 +1,17 @@
 [
+    {
+        "uid": "SOL-2022-7",
+        "name": "StorageWriteRemovalBeforeConditionalTermination",
+        "summary": "Calling Yul functions that conditionally terminate the transaction using ``return(...)`` or ``stop()`` may result in incorrect removals of storage writes.",
+        "description": "A call to a Yul function that conditionally terminates the transaction did not correctly mark all prior changes to storage as 'used', which may result in storage writes to be incorrectly removed. This happens in cases in which it had been valid to remove the store, if the Yul function in question never actually terminated the transaction. Conditional termination of the transaction within the same Yul function is not affected. In Solidity with via-IR code generation, any state change before a function conditionally calling ``return(...)`` or ``stop()`` in inline assembly, may be incorrectly removed, whenever it had been valid to remove the change without the ``return(...)`` or ``stop()``. Since the bug only occurs on calls to Yul functions, inlining may prevent the bug in some cases.",
```

Comment:
Maybe we should add more context on how the optimizer step works?
```suggestion
        "description": "Yul optimizer has a step that detects and removes storage operations which write values that get overwritten before they are ever read. The step tracks all storage writes performed within a Yul function definition or the top-level Yul block and marks each of them as *used*, *unused* or *undecided*. When the control flow jumps to a different function, *undecided* operations should be changed to *used* unless the function is known to always revert. After the analysis, all operations storing unused values can be safely removed without altering the behavior of the contract. For the purpose of the analysis, for each function call it has to be determined whether it can revert, successfully terminate the whole transaction or return back to the caller and continue. More than one of these can be true due to conditionals. A call to a Yul function that conditionally terminates the transaction did not correctly mark all prior changes to storage as 'used', which may result in storage writes to be incorrectly removed. This happens in cases in which it had been valid to remove the store, if the Yul function in question never actually terminated the transaction. Conditional termination of the transaction within the same Yul function is not affected. In Solidity with via-IR code generation, any state change before a function conditionally calling ``return(...)`` or ``stop()`` in inline assembly, may be incorrectly removed, whenever it had been valid to remove the change without the ``return(...)`` or ``stop()``. Since the bug only occurs on calls to Yul functions, inlining may prevent the bug in some cases.",
```
Note that here I'm just adding text before your description, not modifying it.

Reply:
"before they are ever read" doesn't really fit for the storage case, I think... it's rather "if they can never remain visible after the end of the current EVM call" or so...